js/dactylo.js

```diff
@@ -87,7 +87,6 @@ window.addEventListener('DOMContentLoaded', () => {
     const lessonLetters = rawLetters.concat(!has1dk ? [] :
       rawLetters.flatMap(letter => (letter in odk ? [odk[letter]] : [])));
 
-    // XXX we **definitely** need a progress bar here
     gLessonLevel = level;
     gLessonWords = gDictionary.filter(word => {
```

Comment:
Kind of nitpick : I don’t really like writing procedural code in closures, I much prefer staying functional style. I would have written it like this : 
```js
gLessonWords = gDictionary.filter(word =>
  Array.from(word).every(letter => lessonLetters.indexOf(letter) >= 0)
);
```
A string is basically an array, so the performance overhead of `Array.from` should be minimal, right ?

Reply:
This is exactly the part I knew I did wrong. I’m still reluctant to use `Array.from()` for performance considerations but I agree we should default to that.